### ISSUE TITLE
Add support for tests

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#110]
Right now, there is no support in Wasp for testing, and we should look into adding it.

Testing external JS code should not be so hard -> we should set some conventions on how and where to write tests, probably using Jest, and we can then execute them. There might be some unique challenges regarding mocking and similar, but they should be solvable.

Testing Wasp code is different -> since it is aiming to be simple and declarative, unit tests should not be needed, and it would anyway be harder to implement them since it would need to be completely custom solution.
But, e2e tests of the whole app, those are important. We should offer good support for those, probably via Cypress or TestCafe -> this is too complex for us to do on our own, we should just delegate it to existing solutions, but somehow nicely integrate it with Wasp.

So, to recap: We should look into unit testing of JS code via Jest and into e2e testing via Cypress or Testcafe, and find a way to integrate those with Wasp.

**Comment by Martinsos** [comment]
Some ideas that were suggested by community member on Discord:

> Related to testing, I recently read about Testcontainers, not sure how it fits into Wasp but the idea is neat. Can use typescript to create containers running postgres/redis/elasticsearch/etc. and control their lifecycle directly from typescript
>   For even more control over the CI/CD environment, there is also Dagger, but it is more devops-y than Testcontainers which comes with nice defaults
> https://dagger.io/

**Comment by Martinsos** [comment]
We now have frontend testing support! We don't yet have support for backend.
